include/diy/mpi/communicator.hpp

```diff
@@ -58,15 +60,54 @@ namespace mpi
 }
 }
 
+diy::mpi::communicator::
+communicator(MPI_Comm comm)
+:comm_(comm), rank_(0), size_(1)
+{
+#ifndef DIY_NO_MPI
+  if (comm != MPI_COMM_NULL)
+  {
+    MPI_Comm_rank(comm_, &rank_);
+    MPI_Comm_size(comm_, &size_);
+  }
+#endif
+}
+
+diy::mpi::status
+diy::mpi::communicator::
+probe(int source, int tag) const
+{
+  (void) source;
+  (void) tag;
+
+  status s;
+#ifndef DIY_NO_MPI
+  MPI_Probe(source, tag, comm_, &s.s);
+#endif
+  return s;
+}
+
 diy::mpi::optional<diy::mpi::status>
 diy::mpi::communicator::
 iprobe(int source, int tag) const
 {
+  (void) source;
+  (void) tag;
+#ifndef DIY_NO_MPI
   status s;
   int flag;
   MPI_Iprobe(source, tag, comm_, &flag, &s.s);
   if (flag)
     return s;
+#endif
```

Comment:
This seems a bit treacherous to me. Maybe both `probe` and `iprobe` should throw via `DIY_UNSUPPORTED_MPI_CALL`? I can't think of when they'd be used correctly in the no-mpi mode, and especially when returning empty status would be meaningful.

Reply:
sounds good.

Reply:
@mrzv this is not working as expected. iprobe is used in `master::comm_exchange`. On second thought, return a status indicating nothing available here when not built with MPI seems reasonable to me, as that's indeed true. What do you think? Alternatively, I can `ifdef` out the iprobe call in `master::comm_exchange`.